packages/clerk-js/src/ui/UserButton/UserButton.tsx

```diff
@@ -10,29 +10,31 @@ import { UserButtonPopover } from './UserButtonPopover';
 import { UserButtonTrigger } from './UserButtonTrigger';
 
 const _UserButton = () => {
+  const { isOpen: isOpenProp } = useUserButtonContext();
   const { floating, reference, styles, toggle, isOpen } = usePopover({
     defaultOpen: false,
```

Comment:
🔧 I'd expect to pass the `isOpenProp` to `defaultOpen` in `usePopover` hook instead of introducing a new variable. That way we will avoid doing the `isOpenProp || isOpen` check throughout this component.

Reply:
@SokratisVidros That was the initial implementation but it changed. Default only handles the initial state and is not programmatically controlled. 
Something like this `isOpenProp || isOpen` exists because you have to choose how you want to handle the open/close state. Do you want the library to do it or let the user via prop?